crates/storage_models/src/query/generics.rs

```diff
@@ -111,6 +111,66 @@ where
         .attach_printable_lazy(|| format!("Error while updating {debug_values}"))
 }
 
+#[instrument(level = "DEBUG", skip_all)]
+pub async fn generic_update_with_unique_predicate_get_result<T, V, P, R>(
+    conn: &PgPooledConn,
+    predicate: P,
+    values: V,
+) -> StorageResult<R>
+where
+    T: FilterDsl<P> + HasTable<Table = T> + Table + 'static,
+    V: AsChangeset<Target = <<T as FilterDsl<P>>::Output as HasTable>::Table> + Debug + 'static,
+    <T as FilterDsl<P>>::Output: IntoUpdateTarget + 'static,
+    UpdateStatement<
+        <<T as FilterDsl<P>>::Output as HasTable>::Table,
+        <<T as FilterDsl<P>>::Output as IntoUpdateTarget>::WhereClause,
+        <V as AsChangeset>::Changeset,
+    >: AsQuery + LoadQuery<'static, PgConnection, R> + QueryFragment<Pg> + Send,
+    R: Send + 'static,
+{
+    let debug_values = format!("{values:?}");
+
+    let query = diesel::update(<T as HasTable>::table().filter(predicate)).set(values);
+    logger::debug!(query = %debug_query::<Pg, _>(&query).to_string());
```

Comment:
The panic would happen right here if there are no fields to update and not anywhere in the following code.

Can we raise a `NoFieldsToUpdate` error variant if we encounter a `ConnectionError::Query(DieselError::QueryBuilderError(_))`?

I'd prefer it if you used `get_result_async()` instead of `get_results_async()`.

Reply:
> The panic would happen right here if there are no fields to update and not anywhere in the following code.

Yes that would happen. I've mentioned that on line 154. Will handle it. I guess  It would also happen in `generic_update` and `generic_update_with_results` , will check and handle there also.

> I'd prefer it if you used get_result_async() instead of get_results_async().

I doubt if it would throw error or update multiple records and return the first record. Will verify and use accordingly.